values.yaml

```
@@ -834,6 +838,7 @@ server:
       # https://www.vaultproject.io/docs/platform/k8s/helm/run#protecting-sensitive-vault-configurations
       config: |
         ui = true
+        cluster_name = {{ .Values.cluster_name | default (printf "%s-k8s" (include "vault.name" .)) | quote }}
```

You can't set a default here because if someone has already deployed Vault and it has a generated cluster name, this is going to introduce a new set of metrics.
```suggestion
        cluster_name = {{ .Values.cluster_name | default (printf "%s-k8s" (include "vault.name" .)) | quote }}
```

---

`vault.name` is just `vault` right? - so if nothing is set it will always result in the same value no?
If users change it then it may result in what you're describing.

---

I'm referencing Vault's code that generates the cluster name if it's not set in the config. It [generates a cluster name](https://github.com/hashicorp/vault/blob/main/vault/cluster.go#L170-L188) that looks something like this: `vault-cluster-b0435276`. If you now start setting it in the config with a default anyone who was collecting metrics using the Vault generated name will now have a new set metrics.

---

I added defaults in case for new deployment no value is set. I'm open to any proposal that you may be kind enough to commit.